src/main/java/com/urlparser/entity/LinksFromUrl.java

```diff
@@ -0,0 +1,46 @@
+package com.urlparser.entity;
+
+import com.urlparser.model.Links;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class LinksFromUrl implements Links {
+
+  private Document doc;
+  List<URL> listedLinks = new ArrayList<>();
+
+  public LinksFromUrl(String url) {
+    try {
```

Comment:
@undsett _Usually_ it is not a good practice to have any time consuming logic in a constructor.
Instead it is better to save all the prerequisites (like url in this example) and use them at the moment of the actual use.
I'd suggest to move connection and parsing logic to `iterator` method. So we wont connect/parse till the project is used - iterated.
Usually it is ok to be always lazy.

Reply:
agree